torch/testing/_internal/common_methods_invocations.py

```diff
@@ -2082,6 +2082,26 @@ def sample_inputs_index_put(op_info, device, dtype, requires_grad, **kwargs):
 
     return inputs
 
+def sample_inputs_pad(op_info, device, dtype, requires_grad, **kwargs):
+    inputs = [
```

Comment:
partialing make_tensor and appending to inputs incrementally might make this function simpler

what about a sample input with empty padding, too?

Reply:
I'm not sure we can test empty padding here, since OpInfo tests compare results of multiple identical calls. With empty padding, identical calls are free to give different results. I tried adding an empty mode case here and I got a lot of errors for this reason

Reply:
That's a great point. It would still be nice to run some tests, like autograd, for pad with the empty mode.

One option would be to create a second OpInfo for pad where its sample inputs were like:

```
partial(sample_inputs_pad(padding_mode='empty')
```

and then it would skip tests like variant consistency that are trying to verify the op performs the same action across runs. Do you think an approach like that would work or do even tests like autograd rely on predictable outputs?

Reply:
> Do you think an approach like that would work or do even tests like autograd rely on predictable outputs?

I'm not totally sure, I'll check it out when my build finishes.

On a related note, I think it might be a good idea to add a nondeterministic alert for `mode='empty'`, right?

Reply:
Turns out that even the autograd OpInfo tests fail with empty mode

Reply:
Yikes. OK, how about we just add a TODO comment that in the future we'd like to add empty mode.

More generally we should think about aliasing empty to zeros like JAX does.

Does torch.empty itself alert when use_determinstic_algorithms is False?

Reply:
@ngimel suggests that GPU performance would suffer if we aliased empty to zeros, although we could alias empty padding to constant padding with constant value zero in torch.pad if we want to make it deterministic.

What do you think @kurtamohler? Is the perf noticeable in torch.pad, with all its additional operations? Even if the perf is noticeable, is the determinism worth it?

Reply:
We should never alias empty to zeros for performance reasons, and, tbh, there's not much point in adding `empty` mode for padding also, it's not widely used (if used at all).

Reply:
Should we just get rid of empty mode? It does seem impractical

Reply:
Sure; if that would be simpler. We can wait for a user to request it (and maybe no one ever will).

Reply:
I've removed it